### PR TITLE
feat(HACBS-1386): a script to parse hacbs test result

### PR DESCRIPTION
**Pull request by hongweiliu17**
Add a function to parse hacbs test result to generate $HACBS_TEST_OUTPUT
 and log out failed tests

Signed-off-by: Hongwei Liu hongliu@redhat.com

### CODE REVIEW

**Comment by jsztuka** [comment]
Chage looks good overall, do you have some example outputs of running this script?

**Review by jsztuka — APPROVED** [review]
LGTM

**Comment by dirgim** [comment]
@hongweiliu17 the script looks good code-wise, wdyt if it is possible to add some tests for it?

File: test/entrypoint.sh [hunk 1]
```diff
@@ -22,5 +88,13 @@ check_return_code
 # Deprecated Image
 curl -X 'GET' 'https://catalog.redhat.com/api/containers/v1/repositories/registry/registry.access.redhat.com/repository/rhscl%2Fnodejs-8-rhel7' -H 'accept: application/json' > image.json
 
+# Test parse_hacbs_test_output
+test_parse_hacbs_test_output(){
+  conftest test --namespace optional_checks --policy $POLICY_PATH/image/inherited-labels.rego label.json --output=json > unittest.json
+  HACBS_TEST_OUTPUT=
+  parse_hacbs_test_output sanity_label_check conftest unittest.json
+  [ "$(echo $HACBS_TEST_OUTPUT | jq -r '.result')" == "SUCCESS" ] && exit 0 || exit 1
```
Comment (dirgim) [inline test/entrypoint.sh:96]: Looks like calling `exit 0` exits the entrypoint script completely, maybe we can have something like `echo "test_parse_hacbs_test_output PASSED"` instead?

File: test/entrypoint.sh [hunk 2]
```diff
@@ -22,5 +88,13 @@ check_return_code
 # Deprecated Image
 curl -X 'GET' 'https://catalog.redhat.com/api/containers/v1/repositories/registry/registry.access.redhat.com/repository/rhscl%2Fnodejs-8-rhel7' -H 'accept: application/json' > image.json
 
+# Test parse_hacbs_test_output
+test_parse_hacbs_test_output(){
```
Comment (dirgim) [inline test/entrypoint.sh:92]: I think we'll also need to call this function within this file in order to execute it in CI.

File: test/entrypoint.sh (hunk 1 above)
Comment (hongweiliu17) [reply test/entrypoint.sh:96]: Yes, `exit 0` will quit. Thanks!

**Review by dirgim — APPROVED** [review]
LGTM, thanks for adding the tests!

File: test/entrypoint.sh (hunk 2 above)
Comment (hongweiliu17) [reply test/entrypoint.sh]: Yes, thanks.

**Review by srivickynesh — APPROVED** [review]
LGTM

**Comment by hongweiliu17** [comment]
@dirgim I found https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-snyk-check/0.1/sast-snyk-check.yaml#L30-L38 is using feature-sast branch and sast-snyk-check.yaml is using utils.sh, so I think this PR need to be added to https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast, is there a way to cherry-pick, or let me commit create a branch from https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast? Thank you!

**Comment by dirgim** [comment]
> @dirgim I found https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-snyk-check/0.1/sast-snyk-check.yaml#L30-L38 is using feature-sast branch and sast-snyk-check.yaml is using utils.sh, so I think this PR need to be added to https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast, is there a way to cherry-pick, or let me commit create a branch from https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast? Thank you!

@hongweiliu17 that's a good point, we can manually update the `feature-sast` branch and then build and update the image. I can handle that part.

**Comment by hongweiliu17** [comment]
> > @dirgim I found https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-snyk-check/0.1/sast-snyk-check.yaml#L30-L38 is using feature-sast branch and sast-snyk-check.yaml is using utils.sh, so I think this PR need to be added to https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast, is there a way to cherry-pick, or let me commit create a branch from https://github.com/redhat-appstudio/hacbs-test/tree/feature-sast? Thank you!
> 
> @hongweiliu17 that's a good point, we can manually update the `feature-sast` branch and then build and update the image. I can handle that part.

Thank you!

**Pull request merged**